gobblin-yarn/src/main/java/gobblin/yarn/YarnHelixUtils.java

```diff
@@ -199,4 +206,48 @@ public static void addFileAsLocalResource(FileSystem fs, Path destFilePath, Loca
 
     return environmentVariableMap;
   }
-}
+
+  public static Optional<Integer> takePort(int desiredPort) {
```

Comment:
At first I was just getting a random port, but it seemed nicer to get a port that you requested.  The method here is how YourKit works and it feels pretty nice to me because when there are multiple containers on the same node they will all group closely together.

That said, if you are adamant I can change it.